### ISSUE TITLE
timeoutInSeconds on @FindBy doesn't seem to be increasing my timeouts

### ISSUE THREAD

**Issue by dbirch-cognitoiq** [serenity-bdd/serenity-core#987]
I'm seeing some odd behaviour for the timeoutInSeconds parameter on the serenity @FindBy annotation, using serenity, appium and a hybrid app.

I am setting this for an element to 120 seconds, then have tried a number of different waits for the element, but they all seem to be timing out after the default of 2 seconds.

```
@FindBy(id = "appWrapper", timeoutInSeconds = "120")
private WebElementFacade appWrapper
```

Are there specific wait methods that this annotation based timeout approach works with, or am I missing something here? I currently seem to have to use the withTimeoutOf(120, TimeUnit.SECONDS) approach to increase these timeouts, but would really prefer to be able to do this with the annotation rather than every time I access it or have to provide a getter for it which handles this.

I have tried this with serenity 1.2.4 (which I am aware is pretty old), with appium 4.1.2 java bindings. Just trying to get this working in a newer version of serenity to see if it is any better, but the functionality seems to exist in 1.2.4 so I figure I'm missing something!

Not sure what extra detail I can provide here, any advice would be appreciated.

**Comment by dbirch-cognitoiq** [comment]
Some further info, I have updated to serenity 1.4.1-rc.7 and I'm still seeing the same issue. I tried later versions, but they seem to just give me an NPE when I try to run my tests and access the element, will investigate more tomorrow and raise another issue.

Is there anything else I can check to help tie this one down?

**Comment by wakaleo** [comment]
If you can nail down a particular version change, that would be a great help. He Appium integration in Serenity is very much a community effort, so don’t hesitate to dive into the code with the debugger to try to see what is going on as well.

**Comment by dbirch-cognitoiq** [comment]
Any thoughts on the original timeoutInSeconds issue? Are there any cases where you'd expect this annotation to be ignored by design?

I'll do some more investigation now and try and track down the NPE issue and get another issue raised for that.

**Comment by dbirch-cognitoiq** [comment]
After more investigation, the NPE and timeout issues aren't related, have raised a separate issue for the NPE problem.

**Comment by wakaleo** [comment]
The timeoutInSeconds attribute seems to be working correctly for the following test code in the latest version:

    @FindBy(name = "country", timeoutInSeconds="5")
    public WebElementFacade country;

Could you provide a sample project reproducing the issue?

**Comment by dbirch-cognitoiq** [comment]
I will try and get something put together for this.

**Comment by wakaleo** [comment]
A minimal demo app would be sufficient.

**Comment by dbirch-cognitoiq** [comment]
Ok thanks, I shall try and put a quick demo app together and include it with my test code.

**Comment by wakaleo** [comment]
Did you get this sorted out @dbirch-cognitoiq?

**Comment by dbirch-cognitoiq** [comment]
Just got my NPE issue fixed, will retest this on 1.8.2 tomorrow and let you know how I get on!

**Comment by wakaleo** [comment]
Presume fixed
